### PR TITLE
feat: [CCM-5851]: yaml changes for k8s connector and AS rule

### PR DESCRIPTION
**Pull request by rishik30**
##### Summary:

- Changed the YAML content for k8s connector permissions preview
- Removed nginx header from k8s AS rule YAML
- Added support for ingress controller field in K8s AS YAML
- Removed ingress spec from K8s AS YAML

##### Jira Links:

https://harness.atlassian.net/browse/CCM-5851
https://harness.atlassian.net/browse/CCM-5838
https://harness.atlassian.net/browse/CCM-5841
https://harness.atlassian.net/browse/CCM-6863

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by riyasyash — APPROVED** [review]

**Review by jenil-harness — DISMISSED** [review]

**Review by avinashmadhwani02 — DISMISSED** [review]

File: src/modules/35-connectors/components/CreateConnector/CEK8sConnector/ProvidePermissions.tsx [hunk 1]
```diff
@@ -68,21 +70,29 @@ const ProvidePermissions: React.FC<StepProps<StepSecretManagerProps> & ProvidePe
     queryParams: { accountIdentifier: accountId }
   })
 
-  const handleDownload = async () => {
-    trackEvent(CE_K8S_CONNECTOR_CREATION_EVENTS.DOWNLOAD_YAML, {})
+  useEffect(() => {
+    fetchPermissionsYaml()
```
Comment (avinashmadhwani02) [inline src/modules/35-connectors/components/CreateConnector/CEK8sConnector/ProvidePermissions.tsx:74]: Do we need useEffect? can we make use of restful react hook directly?

Comment (rishik30) [reply src/modules/35-connectors/components/CreateConnector/CEK8sConnector/ProvidePermissions.tsx:74]: yes, we can use restful react hook, did not think of that while developing.

Comment (avinashmadhwani02) [reply src/modules/35-connectors/components/CreateConnector/CEK8sConnector/ProvidePermissions.tsx:74]: this way we will not have to store response in the state - restful react is already doing that.

**Comment by rishik30** [comment]
trigger cypress

**Review by avinashmadhwani02 — APPROVED** [review]

**Comment by rishik30** [comment]
trigger cypress

**Review by riyasyash — APPROVED** [review]

**Pull request merged**